### DISCUSSION TITLE
Mixing Singularity and AWS Batch executors

### DISCUSSION THREAD

**Discussion by msneddon** [nextflow-io/nextflow#3380] [Q&A] [answered]
I'm attempting to run a hybrid workflow with most processes executing on a local cluster (with Slurm + singularity) and selected processes running on AWS Batch. I can get each to work separately, but when I mix the two executors I'm running into a problem. I'm using Nextflow version 21.10.5 build 5658.

Instead of executing a process with the AWS Batch executor, it seems to try to interpret the "container" field as a local Singularity container, instead of as a job definition to send to AWS Batch.

For example, with this config snippet (edited slightly to remove real process names) :

```
process.container = "test.sif"

singularity {
    enabled = true
    ...
}

plugins {
    id 'nf-amazon'
}

process {
   withName: testAWS {
        executor = 'awsbatch'
        queue = 'test-batch-queue'
        container = 'job-definition://test-batch-job-definition'
        containerOptions = ''
   }
}
```

I get errors that look like this, as if Nextflow is still trying to run Singularity on our cluster.
```
Error executing process > 'testAWS'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name test-batch-job-definition.img.pulling.1668205500232 job-definition://test-batch-job-definition > /dev/null
  status : 255
  message:
    FATAL:   Unsupported transport type: job-definition
```

Before trying to debug too much, does anyone know if this type of Hybrid execution is supported? Are there ways to mix say Singularity and Docker containers in the same workflow, or customize those settings per process? Since the "singularity.enabled" config looks to be a top level configuration that cannot be customized per process.

Any advice or pointers are appreciated. Thank you!

**Comment by pditommaso** [comment, marked as answer]
This should work using the latest version 22.10.1

**Reply by msneddon** [reply]
Thanks for the quick response. We'll upgrade and try it out!
